crates/wasmparser/src/binary_reader.rs

```diff
@@ -236,10 +236,7 @@ impl<'a> BinaryReader<'a> {
         for _ in 0..returns_len {
             returns.push(self.read_type()?);
         }
-        Ok(FuncType {
-            params: params.into_boxed_slice(),
-            returns: returns.into_boxed_slice(),
-        })
+        Ok(FuncType::new(params, returns))
```

Comment:
Technically we could improve this implementation and reuse allocations but that would have required more code changes and I wanted to keep them minimal.

Reply:
@alexcrichton If you are okay with introducing a `pub(crate) fn FuncType::from_raw_parts(..)` constructor we could implement this micro optimization in this PR. It should reduce the amount of heap allocations while parsing `FuncType`. In practice I was not able to measure a difference. Probably the benchmarks do not have too many big `FuncType` instances.